shogun-lib/src/main/java/de/terrestris/shogun/lib/service/GroupService.java

```diff
@@ -66,90 +59,46 @@ public List<Group> findAllBy(Specification specification) {
         List<Group> groups = (List<Group>) repository.findAll(specification);
 
         for (Group group : groups) {
-            this.setTransientKeycloakRepresentations(group);
+            groupProviderService.setTransientRepresentations(group);
         }
 
         return groups;
     }
 
+    @PostFilter("hasRole('ROLE_ADMIN') or hasPermission(filterObject, 'READ')")
+    public List<User> getGroupMembers(String id) {
+        return groupProviderService.getGroupMembers(id);
+    }
+
     @PostAuthorize("hasRole('ROLE_ADMIN') or hasPermission(returnObject.orElse(null), 'READ')")
     @Transactional(readOnly = true)
     @Override
     public Optional<Group> findOne(Long id) {
         Optional<Group> group = repository.findById(id);
 
         if (group.isPresent()) {
-            this.setTransientKeycloakRepresentations(group.get());
+            groupProviderService.setTransientRepresentations(group.get());
         }
 
         return group;
     }
 
-    @PostFilter("hasRole('ROLE_ADMIN') or hasPermission(filterObject, 'READ')")
-    @Transactional(readOnly = true)
-    public List<Group> findByUser(User user) {
-        List<Group> groups = new ArrayList<>();
-
-        List<GroupRepresentation> keycloakGroups = keycloakUtil.getKeycloakUserGroups(user);
-
-        for (GroupRepresentation keycloakGroup : keycloakGroups) {
-            Optional<Group> group = repository.findByKeycloakId(keycloakGroup.getId());
-            if (group.isPresent()) {
-                group.get().setKeycloakRepresentation(keycloakGroup);
-                groups.add(group.get());
-            }
-        }
-
-        return groups;
-    }
-
-    /**
-     * @deprecated Use KeycloakUtil instead
-     */
-    @Deprecated
-    public GroupRepresentation findByKeycloakId(String keycloakId) {
-        GroupResource groupResource = keycloakUtil.getGroupResource(keycloakId);
-
-        return groupResource.toRepresentation();
-    }
-
-    public List<User> getGroupMembers(String id) {
-        GroupResource groupResource = keycloakUtil.getGroupResource(id);
-        List<UserRepresentation> groupMembers = groupResource.members();
-
-        ArrayList<User> users = new ArrayList<>();
-        for (UserRepresentation groupMember : groupMembers) {
-            Optional<User> user = userRepository.findByKeycloakId(groupMember.getId());
-            if (user.isPresent()) {
-                user.get().setKeycloakRepresentation(groupMember);
-                users.add(user.get());
-            }
-        }
-
-        return users;
-    }
-
     /**
      * Finds a Group by the passed keycloak ID. If it does not exists in the SHOGun DB it gets created.
      *
      * @param keycloakGroupId
      * @return
      */
     @Transactional
-//    @PreAuthorize("hasRole('ROLE_ADMIN') or hasPermission(#keycloakGroupId, 'CREATE')")
+    @PreAuthorize("hasRole('ROLE_ADMIN') or hasPermission(#keycloakGroupId, 'CREATE')")
```

Comment:
if you're wondering why this was commented: I think there was an issue when shogun was out-of-sync with keycloak i.e. when a group existed in keycloak but not in shogun db

if that happens while a normal user is authenticated certain requests may fail because that user doesn't have create permission for `keycloakGroupId`.

Reply:
maybe it's enough to check if the authenticated user can read the group?

Reply:
I think the GroupService / UserService may be invoked by keycloak events: https://github.com/terrestris/shogun/blob/main/shogun-lib/src/main/java/de/terrestris/shogun/lib/listener/KeycloakEventListener.java#L38

In this case there will be no authenticated user

Reply:
Nevermind, I just looked at the new classed again and saw that this is still the same for `KeycloakUserProviderService`